app/pupilfirst_api.rb

```diff
@@ -56,18 +56,18 @@ def valid_status?(status)
     end
 
     def grades_based_on(status)
-      @submission.evaluation_criteria.map do |criteria|
-        {
-          evaluationCriterionId: criteria['id'],
-          grade: grade_for(criteria, status)
-        }
+      if status == 'passed'
+        return @submission.evaluation_criteria.map do |criteria|
+          {
+            evaluationCriterionId: criteria['id'],
+            grade: criteria['max_grade']
+          }
+        end
+      else
+        return []
```

Comment:
I am not sure if we should send in an empty array when we could possibly ignore sending the filed altogether for failed submissions.

Reply:
I was sticking to the way it was written in lms. But I agree. Made the change in lms and grade-action as well.